### DISCUSSION TITLE
404 Page Not Found

### DISCUSSION THREAD

**Discussion by jesperpedersen** [kube-rs/kube#948] [Q&A]
Hi,

I'm currently looking at

```
2022-07-06T08:48:38.812069629-04:00 WARN kube_client::client - Unsuccessful data error parse: 404 page not found
2022-07-06T08:48:38.812085204-04:00 TRACE want - signal: Want
2022-07-06T08:48:38.812127318-04:00 DEBUG kube_client::client - Unsuccessful: ErrorResponse { status: "404 Not Found", message: "\"404 page not found\\n\"", reason: "Failed to parse error data", code: 404 } (reconstruct)
```

which is from the `/apiserver` - but is there a way to get more information on this ? Like is it a missing endpoint (resource) or are credentials to blame.

Getting the full URL plus user info into the `DEBUG` message would be great.

Thanks for any help in advance !

**Comment by nightkr** [comment]
Converting this to an issue (#949) since this is an actionable improvement that can be made to the library.

As a workaround, I believe setting `hyper`'s log level to either `DEBUG` or `TRACE` (sorry, can't remember which) should make it log out all URLs that are requested...

In practice, this error tends to either mean that an API resource is missing (either because K8s is too old or the corresponding CRD hasn't been installed), that the namespace is missing, or that the object you're trying to interact with is missing (if it must exist for the verb you're using to make sense).

**Reply by jesperpedersen** [reply]
Thanks, Teo !

I'll look in that direction (missing resource) - since it is kube-rs 0.73.1 with `kind/main` (k8s 1.24.x)
